doc/development/plugins.rst

```diff
@@ -269,6 +271,82 @@ this may have unintended side-effects and is not recommended.
 .. _installing_plugin:
 
 
+Wire handling
+-------------
+
+PennyLane uses the :class:`~.wires.Wires` class for the internal representation of wires. :class:`~.wires.Wires`
+inherits from python's ``Sequence``, and represents an ordered set of unique wire labels.
+Indexing a ``Wires`` instance will return another ``Wires`` instance of length one.
+The ``labels`` attribute stores a tuple of the wire labels.
+
+For example:
+
+.. code-block:: python
+
+    from pennylane.wires import Wires
+
+    wires = Wires(['ancilla', 0, 1])
+    print(wires[0]) # <Wires = ['ancilla']>
+    print(wires.labels) # ('ancilla', 0, 1)
+
+As shown in the section on :doc:`/introduction/circuits`, a device can be created with custom wire labels:
+
+.. code-block:: python
+
+    from pennylane import *
+
+    dev = device('my.device', wires=['q11', 'q12', 'q21', 'q22'])
+
+    @qnode(dev)
+    def circuit():
+       MyGate1(wires='q22')
+       MyGate2(wires=['q21','q11'])
+       MyGate1(wires=['q21'])
+       return expval(MyObs(wires='q11')@ MyObs(wires='q12'))
+
+Behind the scenes, when ``my.device`` gets created it turns ``['q11', 'q12', 'q21', 'q22']`` into a
+:class:`~.wires.Wires` object and stores it in the device's ``wires`` attribute. Likewise, when gates and
+observables get created they turn their ``wires`` argument into a :class:`~.wires.Wires`
+object and store it in their ``wires`` attribute.
+
+.. code-block:: python
+
+    print(dev.wires) #  <Wires = ['q11', 'q12', 'q21', 'q22']>
+
+    op = MyGate2(wires=['q21','q11'])
+    print(op.wires) # <Wires = ['q21', 'q11']>
+
+When the device applies operations, it needs to translate
+``op.wires`` into wire labels that the backend "understands". This can be done with the
+:meth:`~._device.Device.map_wires` method.
+
+.. code-block:: python
+
+    # inside the 'my.device' class inheriting from the base device class
+    device_wires = self.map_wires(op.wires)
```

Comment:
One thing that took me a while to understand was that `device_wires` was itself a `Wires` object, just with the mapped indices (I was expecting a list) - this is good, just thought it might be nice to clarify